app/uk/gov/hmrc/helptosavefrontend/connectors/HelpToSaveConnector.scala

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavefrontend.connectors
+
+import java.net.URLEncoder
+
+import cats.data.EitherT
+import cats.instances.future._
+import com.google.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import uk.gov.hmrc.helptosavefrontend.config.{WSHttp, WSHttpExtension}
+import uk.gov.hmrc.helptosavefrontend.models.{EligibilityResult, UserInfo}
+import uk.gov.hmrc.helptosavefrontend.util.{NINO, Result}
+import uk.gov.hmrc.helptosavefrontend.util.HttpResponseOps._
+import uk.gov.hmrc.play.config.ServicesConfig
+import uk.gov.hmrc.play.http.{HeaderCarrier, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+trait  HelpToSaveConnector {
+
+  def getEligibilityStatus(nino: NINO, userDetailsURI: String)(implicit hc: HeaderCarrier): Result[EligibilityResult]
+
+  def createAccount(userInfo: UserInfo)(implicit hc: HeaderCarrier): Result[UserInfo]
+}
+
+@Singleton
+class HelpToSaveConnectorImpl @Inject()(implicit ec: ExecutionContext) extends HelpToSaveConnector with ServicesConfig{
+
+  val helpToSaveUrl: String = baseUrl("help-to-save")
+
+  val createAccountURL: String = s"$helpToSaveUrl/create-an-account"
+
+  def eligibilityURL(nino: NINO, userDetailsURI: String): String =
+    s"$helpToSaveUrl/eligibility-check?nino=$nino&userDetailsURI=${URLEncoder.encode(userDetailsURI, "UTF-8")}"
+
+  def badResponseMessage(response: HttpResponse, description: String): String =
```

Comment:
can `description` be renamed as `service` ? so that the message reading will be `$service call returned.....`